method-info/totals-and-components/totals-and-components.md

```diff
@@ -0,0 +1,115 @@
+able to call the method on a set of data. The tcc markers returned determine
+if and what the method has corrected.
+
+* "S" is when the method stops
+* "N" is for no correction
+* "M" is for manual editing this is where the record needs to be reviewed by a user
+* "C" is for component correction where the final components are corrected
+* "T" is for totals correction where the final totals are corrected
+
+The core mathematical corrections are total correction and component correction shown below respectively from the methodology specification.
+
+```bash
+    final_total = sum_of_components
+```
+
+and 
+
+```bash
+    final_component = (original_component / sum_of_components) * total
+```
+
+It is worth noting these are mutually exclusive and a result of many mathematical and logical steps. Therefore, to get a complete understanding of the above equations it is advisable to read the methodical or technical specifications.
+
+From the methodology and technical specification we can see that the method accepts a structured input.
+The input parameters determine how the method operates and what kind of outputs we would expect from it.
+
+### Example Run Through
+
+Below is a snapshot of an example dataset and how the input data should
+look like:
+
+| identifier | total | components | amend_total | predictive | precision | auxiliary | absolute_difference_threshold | percentage_difference_threshold
+| --- | --- | --- | --- | --- | --- | --- | --- | --- |
+| 1 | 1689 | [632,732,101,165] | False | 1689 | 10 | None | 28 | 0.1 |
+| 2 | 0 | [7,0,2,2] | True | 0 | 28 | None | 11 | None |
+| 3 | 11 | [0,0,0,0] | False | 11 | 28 | None | 11 | None |
+| 4 | 10811 | [9201,866,632,112] | True | 10811 | 28 | None | None | 0.1 |
+| 5 | 12492 | [9201,866,632,112] | True | 12492 | 28 | None | None | 0.1 |
+
+This is constructed from the following:
+
+* Unique Identifier – Any e.g., Business Reporting Unit
+* Total Variable – Target period total, numeric
+* Components Variable – Corresponding list of Total variable's components,
+ numeric – nulls allowed
+* Amend Total – Select whether Total Variable for the target period should be
+ automatically corrected, Boolean
+* Predictive Variable – Previous or current period total, numeric
+* Precision - The precision value determines the level of accuracy for our floating point calculations
+* Auxiliary Variable – optional, numeric – nulls allowed
+* Absolute Difference Threshold - represented as a decimal
+* Percentage Difference Threshold - represented as a decimal
+
+To run the method using the example above you can use an IDE and create a python file as follows.
+
+```python
+# Importing the totals_and_components method from the totals_and_components.py file
+from sml_small.editing.totals_and_components import totals_and_components
+# The data we are going to pass into the T&C method
+data = ["1", 1689, [(632), (732), (101), (165)], False, 1689, 10, None, 28, 0.1]
```

Comment:
Would it be better to change this into a dictionary and pass it in as **kwargs rather than *args for clarity? something like 
data={
"identifier": 1,
"total": 1689,
etc
etc
}

Reply:
I will ask @sanjeevz3009  on this one as its extracted from example.py so may need updating across the board?

Reply:
Sanjeev talked to Steve and we think its better to leave it as this for now. Thanks for the tip though Jason :)